src/devices/Pn5180/Pn5180.cs

```diff
@@ -0,0 +1,1496 @@
+        /// Byte 1 = Register Address
+        /// next 4 bytes = data of the register
+        /// </summary>
+        public const int RadioFrequenceConfiguraitonSize = 5;
+
+        /// <summary>
+        /// PN532 SPI Clock Frequency
+        /// </summary>
+        public const int SpiClockFrequency = 7_000_000;
+
+        /// <summary>
+        /// Only SPI Mode supported is Mode0
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode0;
+
+        /// <summary>
+        /// The Log level
+        /// </summary>
+        public LogLevel LogLevel
+        {
+            get { return LogInfo.LogLevel; }
+            set { LogInfo.LogLevel = value; }
+        }
+
+        /// <summary>
+        /// The location to log the info
+        /// </summary>
+        public LogTo LogTo
+        {
+            get { return LogInfo.LogTo; }
+            set { LogInfo.LogTo = value; }
+        }
+
+        /// <summary>
+        /// Create a PN5180 RFID/NFC reader
+        /// </summary>
+        /// <param name="spiDevice">The SPI device</param>
+        /// <param name="pinBusy">The pin for the busy line</param>
+        /// <param name="pinNss">The pin for the SPI select line. This has to be handle differently than thru the normal process as PN5180 has a specific way of working</param>
+        /// <param name="gpioController">A GPIO controller, null will use a default one</param>
+        /// <param name="shouldDispose">Dispose the SPI and the GPIO controller at the end if true</param>
+        /// <param name="logLevel">The log level</param>
+        public Pn5180(SpiDevice spiDevice, int pinBusy, int pinNss, GpioController gpioController = null, bool shouldDispose = true, LogLevel logLevel = LogLevel.None)
+        {
+            LogLevel = logLevel;
+
+            LogInfo.Log($"Opening PN5180, pin busy: {pinBusy}, pin NSS: {pinNss}", LogLevel.Debug);
+            _spiDevice = spiDevice;
+            _gpioController = gpioController ?? new GpioController(PinNumberingScheme.Logical);
+            _shouldDispose = shouldDispose;
+            _pinBusy = pinBusy;
+            _pinNss = pinNss;
+            _gpioController.OpenPin(_pinBusy, PinMode.Input);
+            _gpioController.OpenPin(_pinNss, PinMode.Output);
+
+            // Check the version
+            var (product, firmware, eeprom) = GetVersion();
+            if ((product.Major == 0) || (firmware.Major == 0) || (eeprom.Major == 0))
+            {
+                throw new IOException($"Not a valid PN5180");
+            }
+        }
+
+        /// <summary>
+        /// Dispose
+        /// </summary>
+        public void Dispose()
+        {
+            // If the RF field is on, try to switch it off
+            try
+            {
+                SetRadioFrequence(false);
+            }
+            catch (Exception)
+            {
+                // We don't do anything on purpose. Trying to switch off
```

Comment:
when would above throw? can we make it not throw and i.e. return some kind of status or bool flag? (i.e. TrySetRadioFrequency)

Reply:
it's actually obsolete, I can remove fully the try/catch. I changed it for the SetRadioFrequency to return a boolean which in this case we won't have to test.